web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1129,34 +1134,55 @@ namespace com.keyman.keyboards {
         }
       } else if(options['context'] == 'language') { // Download the full list of supported keyboard languages
         this.languageList = x['languages'];
-        if(this.languagesPending) {
-          this.addLanguageKeyboards(this.languagesPending);
-        }
-        this.languagesPending = [];
       }
 
-      return null; // TODO (#5044): return the list of newly-registered stubs here.
+      return this.keyboardStubs.slice(currentKeyboardStubsCount);
     }
 
     /**
      *  Internal handler for processing keyboard registration, used only by `register`
      *
-     *  @param  {Object}   languages    Array of language names
+     *  @param  {string[]}   languages    Array of language names
+     *  @returns {Promise<KeyboardStub[]>} Promise of added keyboard stubs
      **/
-    addLanguageKeyboards(languages) {
-      var i, j, lgName, cmd, first, addAll;
+    addLanguageKeyboards(languages: string[]): Promise<KeyboardStub[]> {
+      var i, j, lgName, cmd, addAll, retPromise;
 
       // Defer registering keyboards by language until the language list has been loaded
-      if(this.languageList == null) {
-        first = (this.languagesPending.length == 0);
-
-        for(i=0; i<languages.length; i++) {
-          this.languagesPending.push(languages[i]);
+      if (this.languageList == null) {
+        if (this.firstCall) {
```

Comment:
Can we refactor to avoid this variable? split this into two functions?

Reply:
Split into issue #5367